src/callbacks/callback_dump_activations.cpp

```diff
@@ -34,23 +32,26 @@ namespace lbann {
 
 void lbann_callback_dump_activations::on_forward_prop_end(model *m, Layer *l) {
 
-  //Skip if we are interested in saving inferences at (a) given layer(s)
-  if(!m_layer_names.empty()) return;
+  // Skip if we are interested in saving inferences at (a) given layer(s)
+  if (!m_layer_names.empty()) { return; }
 
-  // Skip target layers
-  if (dynamic_cast<generic_target_layer*>(l) != nullptr) {
-    return;
```

Comment:
Why eliminate this check?

Reply:
We now iterate through all the activations rather than explicitly dumping the first activation. Target layers have no children, so it doesn't have any activations to dump.